drop-boxes/register-omero-metadata/register-omero.py

```diff
@@ -159,68 +236,121 @@ def process(transaction):
 		#registrationProcess.extractMetadataFromTSV()
 
+		properties = getPropertyMap(line, property_names)
+		print "Metadata properties:\t"
+		printPropertyMap(properties)
+		
+		#one file can have many images, iterate over all img ids
+		for img_id in omero_image_ids:
+			registrationProcess.registerKeyValuePairs(img_id, filterOmeroPropertyMap(properties))
+
+
+		####
 		# 6. In addition to the image registration and technical metadata storage, we want to add
 		# further experimental metadata in openBIS. This metadata contains information about the 
 		# imaging experiment itself, such as modality, imaged tissue and more. 
 		# We also want to connect this data with the previously created, corresponding OMERO image id t
 		# hat represents the result of this experiment in OMERO. 
-		#registrationProcess.registerExperimentDataInOpenBIS(omero_image_ids)
-
-		# 7. Last but not least we create the open science file format for images which is
-		# OMERO-Tiff and store it in OMERO next to the proprierary vendor format.
-		#registrationProcess.triggerOMETiffConversion()
+		#registrationProcess.registerExperimentDataInOpenBIS(omero_image_ids) # I did it myyy wayyyy
 
-		####################
-
-		# TODO decide if new experiment is needed based on some pre-defined criteria.
+		# we decide if new experiment is needed based on some pre-defined criteria.
 		# Normally, the most important criterium is collision of experiment type properties
 		# between samples. E.g. two different imaging modalities need two experiments.
 		
-		#fileBelongsToExistingExperiment = isSameExperimentMetadata(previousProps, properties)
-		#previousProps = properties
-		#if(not fileBelongsToExistingExperiment):
-		#	exp = createNewImagingExperiment(transaction, space, project_code, properties)
-		#imagingSample = createNewImagingRun(transaction, sa, exp, list_of_omero_ids, offset)# maybe there are sample properties, too!
-		# register the actual data
-		#IMAGING_DATASET_CODE = Q_BMI_GENERIC_IMAGING_DATA # I guess
-		#dataset = transaction.createNewDataSet(IMAGING_DATASET_CODE)
-		#dataset.setSample(imagingSample)
-		#transaction.moveFile(imageFile, dataset)
-		# increment id offset for next sample in this loop - not sure anymore if this is needed
-		
\ No newline at end of file
+		fileBelongsToExistingExperiment = isSameExperimentMetadata(previousProps, properties)
+		previousProps = properties
+		if(not fileBelongsToExistingExperiment):
+			imagingExperiment = createNewImagingExperiment(transaction, space, project_code, properties, existing_experiment_ids)
+		imagingSample = createNewImagingRun(transaction, tissueSample, imagingExperiment, omero_image_ids, image_number, properties)
+		# increment id offset for next sample in this loop
+		image_number += 1
+
+		# 7. Last but not least we create the open science file format for images which is
+		# OMERO-Tiff and store it in OMERO next to the proprierary vendor format.
+		#registrationProcess.triggerOMETiffConversion()
```

Comment:
```suggestion
```

Reply:
I would leave it as is, something we can add or remove in a following version

Reply:
Then I would suggest to leave it out and to introduce it on the development branch

Reply:
I don't follow, u mean remove the line for the release but leave it in for the dev. branch?

Reply:
would be hard to do that when we merge back, so maybe create a new branch from dev for that feature and implement it whenever?